extras/corelens.py

```diff
@@ -0,0 +1,140 @@
+                        vmcore_path = value
+
+            if vmcore_path:
+                if storage and (storage.startswith("LABEL=") or storage.startswith("UUID=")):
+                    resolved_storage = subprocess.getoutput(f"findmnt -rn -o TARGET -S {storage}")
+                    if resolved_storage:
+                        storage = resolved_storage
+                        return f"{storage}/{vmcore_path}" if not vmcore_path.startswith("/") else f"{storage}{vmcore_path}"
+                return vmcore_path
+            else:
+                return None
+        except Exception:
+            return None
+
+    def process_recent_vmcores(self, days):
+        # This section is is to check the directory list under the path specified in kdump.conf
+        # looks only for latest 5 directories
+        error_logs = []
+        try:
+            vmcore_dir_path = self.get_vmcore_dir_path()
+            if not vmcore_dir_path:
+                error_logs.append("Error: Could not determine vmcore path from /etc/kdump.conf")
+                self.add_string_as_file("\n".join(error_logs), filename="corelens")
+                return
+
+            vmcore_dir_path = Path(vmcore_dir_path)
+            if not vmcore_dir_path.is_dir():
+                error_logs.append(f"Error: {vmcore_dir_path} is not a valid directory.")
+                self.add_string_as_file("\n".join(error_logs), filename="corelens")
+                return
+
+            now = datetime.now()
+            oldest_allowed_date = now - timedelta(days)
+            recent_vmcore_dirs = []
+
+            for directory in vmcore_dir_path.iterdir():
+                if directory.is_dir():
+                    try:
+                        timestamp_str = directory.name[-19:]
+                        dir_timestamp = datetime.strptime(timestamp_str, "%Y-%m-%d-%H:%M:%S")
+                        if dir_timestamp >= oldest_allowed_date:
+                            recent_vmcore_dirs.append((dir_timestamp, directory))
+                    except ValueError:
+                        error_logs.append(f"Warning: Skipping invalid directory format: {directory.name}")
+
+            recent_vmcore_dirs.sort(reverse=True, key=lambda x: x[0])
+            recent_vmcore_dirs = [directory[1] for directory in recent_vmcore_dirs[:3]]
+            if not recent_vmcore_dirs:
+                error_logs.append("No recent vmcore directories found.")
+                self.add_string_as_file("\n".join(error_logs), filename="corelens")
+                return
+
+            archive_tmp_dir = self.archive.get_tmp_dir()
+            corelens_output_path = Path(archive_tmp_dir) / "sos_commands/corelens"
+            corelens_output_path.mkdir(parents=True, exist_ok=True)
+
+            kernel_version = subprocess.check_output("uname -r", shell=True, universal_newlines=True).strip()
+            vmlinux_ctf_path = f"/usr/lib/modules/{kernel_version}/kernel/vmlinux.ctfa"

```

Comment:
You're saying here that the kernel version for this vmcore must be the same as the current kernel. This may not be true. 

I'd recommend not specifying the `--ctf` parameters for corelens. It will determine the kernel version automatically and select CTF (or DWARF) if available.